src/main/java/io/pinecone/exceptions/HttpErrorMapper.java

```diff
@@ -0,0 +1,22 @@
+package io.pinecone.exceptions;
+
+public class HttpErrorMapper {
+
+    public static void mapHttpStatusError(FailedRequestInfo failedRequestInfo) {
+        int statusCode = failedRequestInfo.getStatus();
+        switch (statusCode) {
+            case 400:
+                throw new PineconeBadRequestException(failedRequestInfo.getMessage());
+            case 401:
+                throw new PineconeAuthorizationException(failedRequestInfo.getMessage());
+            case 404:
+                throw new PineconeNotFoundException(failedRequestInfo.getMessage());
+            case 409:
+                throw new PineconeAlreadyIndexExistsException(failedRequestInfo.getMessage());
```

Comment:
```suggestion
                throw new PineconeConflictException(failedRequestInfo.getMessage());
```

I think a 409 error could also result from trying to create other named resources such as Collections (and maybe more in the future. So we shouldn't put `Index` in the exception name IMO.

Reply:
Agreed!